File: Classes/LogRegCode/logReg.m
function ww = logReg(XX,yy)
[NN, DD]=size(XX);
XX=[ones(NN,1) XX];

ww=zeros(DD+1,1);

for iter=1:1000
    mu = 1./(1+exp(-XX*ww));
    S=diag(mu.*(1-mu));
    ww = ww + inv(XX'*S*XX)*XX'*(yy-mu);
end

File: Classes/LogRegCode/logReg.py
import numpy as np

def logRegF(XX, yy):
    NN, DD = np.shape(XX)
    xdata = np.concatenate((np.ones((NN,1)), XX), axis=1)
    ww = np.zeros((DD+1,1))
    for i in range(1000)    :
        mu = 1/(1+np.exp(-np.dot(xdata,ww)))    
        S = np.diag(np.multiply(mu,(1-mu))[:,0])
        aux = np.linalg.inv(np.dot(np.dot(xdata.T,S),xdata))
        aux = np.dot(np.dot(aux,xdata.T), yy-mu)
        ww = ww + aux
    return ww 

